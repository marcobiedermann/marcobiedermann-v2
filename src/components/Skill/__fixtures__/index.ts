const skill = {
  id: '327e5c73-79e2-45b2-be19-667905bc37de',
  title: 'JavaScript',
};

export default skill;
