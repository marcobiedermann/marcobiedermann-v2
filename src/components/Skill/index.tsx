import { documentToReactComponents } from '@contentful/rich-text-react-renderer';
import { Document } from '@contentful/rich-text-types';
import React from 'react';

export interface SkillProps {
  description?: {
    json: Document;
  };
  id: string;
  title: string;
}

const Skill: React.FC<SkillProps> = (props) => {
  const { description, title } = props;

  return (
    <div>
      <h3>{title}</h3>
      <div>{description && documentToReactComponents(description.json)}</div>
    </div>
  );
};

export default Skill;
