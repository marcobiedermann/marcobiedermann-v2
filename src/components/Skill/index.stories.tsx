/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Skill from '.';
import skill from './__fixtures__';

storiesOf('Skill', module).add('default', () => <Skill {...skill} />);
