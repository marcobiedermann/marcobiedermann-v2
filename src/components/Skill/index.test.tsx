/* eslint-disable import/no-extraneous-dependencies, @typescript-eslint/no-unused-vars */

import React from 'react';
import renderer from 'react-test-renderer';
import Skill from '.';
import skill from './__fixtures__';

describe('Skill component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Skill {...skill} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
