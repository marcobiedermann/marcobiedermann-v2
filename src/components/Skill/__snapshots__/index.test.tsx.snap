// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill component should render correctly 1`] = `
<div>
  <h3>
    JavaScript
  </h3>
  <div />
</div>
`;
